File: Inverted-Index/src/keywords/Ordinary.java
package keywords;
import java.util.Set;

import file_handler.Doc;

public class Ordinary implements Keyword {

    private String word;

    public Ordinary(String word){
        this.word = word;
    }

    public String getWord() {
        return word;
    }

    @Override
    public Set<Doc> operate(Set<Doc> docs, Set<Doc> newDocs) {

        if(docs == null) {
            return newDocs;
        }
        if(newDocs != null)
            docs.retainAll(newDocs);
        return docs;
    }
}

File: Phase5-TDD/InvertedIndex/Keywords/Ordinary.cs
using System;
using System.Collections.Generic;
using InvertedIndex.FileHandler;

namespace InvertedIndex.Keywords
{
    public class Ordinary : Keyword
    {
        private string word;

        public Ordinary(string word)
        {
            this.word = word;
        }

        public string GetWord()
        {
            return word;
        }
        public HashSet<Doc> Operate(HashSet<Doc> docs, HashSet<Doc> newDocs)
        {
            if(docs == null){
                return newDocs;
            }
            if(newDocs != null){
                docs.IntersectWith(newDocs);
            }
            return docs;
        }
    }
}